src/top/utop.el

```diff
@@ -635,10 +656,18 @@ it is started."
      ;; Split lines. Each line contains exactly one command
      (let ((lines (split-string utop-output "\n")))
        (while (>= (length lines) 2)
-         ;; Process the first line
-         (utop-process-line (car lines))
-         ;; Remove it and continue
-         (setq lines (cdr lines)))
+         ;; process first line. if we haven't tried reading the version, we'll
+         ;; trying to do it now.
+         (let ((line (car lines)))
+           (if utop--read-version
+               (utop-process-line line)
+             (progn
+               (save-match-data
+                 (when (string-match "protocol-version:\\([0-9]+\\)" line)
```

Comment:
If the line doesn't match, do we still process it normally or does it end up being discarded?

Reply:
Good point. I'll fix that.